cpp/src/arrow/flight/internal.cc

```diff
@@ -251,7 +262,11 @@ static grpc::Status ToRawGrpcStatus(const Status& arrow_status) {
   } else if (arrow_status.IsAlreadyExists()) {
     grpc_code = grpc::StatusCode::ALREADY_EXISTS;
   }
-  return grpc::Status(grpc_code, message);
+  if (flight_status && !flight_status->ExtraInfo().empty()) {
+    return grpc::Status(grpc_code, message, flight_status->ExtraInfo());
```

Comment:
Does it hurt to always just set ExtraInfo even if it's blank?

Reply:
removed, I dont think this is what the function `ToRawGrpcStatus` should be doing